src/main/java/com/google/api/codegen/config/SampleSpec.java

```diff
@@ -51,7 +49,15 @@
   public SampleSpec(MethodConfigProto methodConfigProto) {
     specified = methodConfigProto.hasSamples();
     sampleConfiguration = methodConfigProto.getSamples();
-    storeValueSets(methodConfigProto.getSampleValueSetsList(), methodConfigProto.getName());
+    valueSets = methodConfigProto.getSampleValueSetsList();
+
+    HashSet<String> ids = new HashSet<>();
+    for (SampleValueSet valueSet : valueSets) {
+      String id = valueSet.getId();
+      if (!ids.add(id)) {
+        throw new IllegalArgumentException("duplicate element: " + id);
```

Comment:
Include the method name here to make it easier for users to fix the config when this exception is thrown.

Reply:
Done